pkg/image/cdi/validate.go

```diff
@@ -67,3 +87,30 @@ func (cv *Validator) Update(oldVMImg, newVMImg *harvesterv1.VirtualMachineImage)
 func (cv *Validator) Delete(_ *harvesterv1.VirtualMachineImage) error {
 	return nil
 }
+
+func (cv *Validator) checkVolumeModeAccessModes(vmImg *harvesterv1.VirtualMachineImage) error {
+	if vmImg.Spec.TargetStorageClassName == "" {
+		return werror.NewInvalidError("targetStorageClassName is required", "spec.targetStorageClassName")
+	}
+
+	// storage profile name is always the same as storage class name
+	sp, err := cv.spCache.Get(vmImg.Spec.TargetStorageClassName)
+	if err != nil {
+		return werror.NewInvalidError(fmt.Sprintf("failed to get storage profile %q: %v", vmImg.Spec.TargetStorageClassName, err), "spec.targetStorageClassName")
+	}
+
+	if len(sp.Spec.ClaimPropertySets) == 0 {
+		sc, err := cv.scCache.Get(vmImg.Spec.TargetStorageClassName)
+		if err != nil {
+			return werror.NewInvalidError(fmt.Sprintf("failed to get storage class %q: %v", vmImg.Spec.TargetStorageClassName, err), "spec.targetStorageClassName")
+		}
+		if _, found := cdicaps.GetCapabilities(cv.client, sc); !found {
+			return werror.NewInvalidError(
+				fmt.Sprintf("no default volume mode or access mode, please specify %s annotation in storage class %q",
+					util.AnnotationStorageProfileVolumeModeAccessModes, vmImg.Spec.TargetStorageClassName),
+				"spec.targetStorageClassName")
+		}
+	}
+
+	return nil
```

Comment:
suggest removing the nested if statement:
```suggestion
	if len(sp.Spec.ClaimPropertySets) != 0 {
		return nil
	}

	sc, err := cv.scCache.Get(vmImg.Spec.TargetStorageClassName)
	if err != nil {
		return werror.NewInvalidError(fmt.Sprintf("failed to get storage class %q: %v", vmImg.Spec.TargetStorageClassName, err), "spec.targetStorageClassName")
	}
	if _, found := cdicaps.GetCapabilities(cv.client, sc); !found {
		return werror.NewInvalidError(
			fmt.Sprintf("no default volume mode or access mode, please specify %s annotation in storage class %q",
				util.AnnotationStorageProfileVolumeModeAccessModes, vmImg.Spec.TargetStorageClassName),
			"spec.targetStorageClassName")
	}

	return nil
```